src/modules/flow/iio/nodes.c

```diff
@@ -25,43 +25,117 @@
 
 #include <sol-iio.h>
 
-struct gyroscope_data {
+enum iiodevice_types {
+    IIO_GYROSCOPE=1,
+    IIO_MAGNETOMETER=2,
+    IIO_THERMOMETER=3,
+    IIO_PRESSURE_SENSOR=4,
+    IIO_COLOR=5,
+    IIO_ACCELEROMETER=6,
+    IIO_HUMIDITY_SENSOR=7,
+    IIO_ADC=8,
+    IIO_LIGHT_SENSOR=9,
+    IIO_PROXIMITY=10,
+};
+
+struct iiodevice_out {
+    union {
+        struct sol_direction_vector direction;
+        struct sol_drange drange;
+        struct sol_rgb rgb;
+    };
+};
+
+#define MAX_CHANNEL			(8)
+
+struct iiodevice_data {
     struct sol_iio_config config;
-    struct sol_direction_vector scale;
-    struct sol_direction_vector offset;
+    union {
+        struct sol_direction_vector scale_vector;
+        double scales[MAX_CHANNEL];
+    };
+
+    union {
+        struct sol_direction_vector offset_vector;
+        double offsets[MAX_CHANNEL];
+    };
```

Comment:
I don't get this unions. Why? I'd say you only need `offsets` and `scales`. The only place where you set `offset_vector` or `scale_vector` is on gyroscope code, but you override it nonetheless when add channels...